src/pages/iou/steps/NewRequestAmountPage.js

```diff
@@ -0,0 +1,211 @@
+import styles from '../../../styles/styles';
+import HeaderWithBackButton from '../../../components/HeaderWithBackButton';
+import ScreenWrapper from '../../../components/ScreenWrapper';
+
+const propTypes = {
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            iouType: PropTypes.string,
+            reportID: PropTypes.string,
+        }),
+    }),
+
+    /** The report on which the request is initiated on */
+    report: reportPropTypes,
+
+    /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
+    iou: PropTypes.shape({
+        id: PropTypes.string,
+        amount: PropTypes.number,
+        currency: PropTypes.string,
+        participants: PropTypes.arrayOf(
+            PropTypes.shape({
+                accountID: PropTypes.number,
+                login: PropTypes.string,
+                isPolicyExpenseChat: PropTypes.bool,
+                isOwnPolicyExpenseChat: PropTypes.bool,
+                selected: PropTypes.bool,
+            }),
+        ),
+    }),
+
+    // eslint-disable-next-line react/forbid-prop-types
+    errors: PropTypes.object,
+};
+
+const defaultProps = {
+    route: {
+        params: {
+            iouType: '',
+            reportID: '',
+        },
+    },
+    report: {},
+    iou: {
+        id: '',
+        amount: 0,
+        currency: CONST.CURRENCY.USD,
+        participants: [],
+    },
+    errors: {},
+};
+
+function NewRequestAmountPage({route, iou, report, errors}) {
+    const {translate} = useLocalize();
+
+    const prevMoneyRequestID = useRef(iou.id);
+    const textInput = useRef(null);
+
+    const iouType = lodashGet(route, 'params.iouType', '');
+    const reportID = lodashGet(route, 'params.reportID', '');
+    const isEditing = lodashGet(route, 'path', '').includes('amount');
+    const currentCurrency = lodashGet(route, 'params.currency', '');
+
+    const currency = currentCurrency || iou.currency;
+
+    /**
+     * Focus text input
+     */
+    const focusTextInput = () => {
+        // Component may not be initialized due to navigation transitions
+        // Wait until interactions are complete before trying to focus
+        InteractionManager.runAfterInteractions(() => {
+            // Focus text input
+            if (!textInput.current) {
+                return;
+            }
+
+            textInput.current.focus();
+        });
+    };
+
+    useFocusEffect(
+        useCallback(() => {
+            focusTextInput();
+        }, []),
+    );
+
+    // Check and dismiss modal
+    useEffect(() => {
+        if (!ReportUtils.shouldHideComposer(report, errors)) {
+            return;
+        }
+        Navigation.dismissModal(reportID);
+    }, [errors, report, reportID]);
+
+    // Because we use Onyx to store iou info, when we try to make two different money requests from different tabs, it can result in many bugs.
```

Comment:
>it can result in many bugs.

NAB, it would be better to say which ones if possible. People won't know which bugs are fixed - and won't think of any alternative solutions or be able to test if the logic works as intended or can be removed in the future, etc.

The code here is still not very intuitive or explained well.